### ISSUE TITLE
Alternative Text for Images in titles

### ISSUE THREAD

**Issue by rachfop** [facebook/docusaurus#8393]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Looks like there is a bug with how docusaurus renders  alternate texts placed in titles.
But if you have a title with an image and alias:
`## ![Run the Application](/img/icons/run.png) Run the Application`
The title is duplicated in the sidebar.
This unfortunate because images should have alternate text (alt text) for accessibility.

### Steps to reproduce

1. Create a title with an svg or image
2. Provide alt text in the image
3. Look at the side bar

### Expected behavior

Alt text shouldn't appear in the side bar.

### Actual behavior

<img width="312" alt="Screenshot 2022-11-29 at 2 06 40 PM" src="https://user-images.githubusercontent.com/64233065/204658746-c25c3919-84c9-48ec-a1ab-a53461973680.png">

### Your environment

- Public source code:
- Public site URL:
- Docusaurus version used:
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):
 

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
You do not need alt messages if the image is purely decorative (which is the case here—it's an icon). This is especially the case if your alt text is duplicated with the text content—it always signals _poor_ accessibility.

Imagine two cases. First one is where you are reading the page with a screen reader. When reading the title, the reader will tell you there's an image and its message, and then immediately followed by the title text itself. If that's less familiar, consider when you disable images in your browser or when the image is broken. The alt message will be shown in place of the image—which means you end up with duplicated text anyway.

TL;DR: "always add alt messages helps with accessibility" is a myth; adding alt messages where there should be none is a net minus to accessibility. Moreover, images in headings should always be decorative, otherwise the page structure is really weird. All this means our behavior is intended.
